server/src/main/java/org/opensearch/search/startree/filter/provider/StarTreeFilterProvider.java

```diff
@@ -156,4 +163,176 @@ public StarTreeFilter getFilter(SearchContext context, QueryBuilder rawFilter, C
 
     }
 
+    class BoolStarTreeFilterProvider implements StarTreeFilterProvider {
```

Comment:
We can move this to a separate source file.

Reply:
Done